### PR TITLE
AirStripping0D and AirWaterEquilibrium Property Model + Docs

### PR DESCRIPTION
**Pull request by kurbansitterley**
This PR will add:

- AirWaterEquilibrium (AWE) property model and related tests
- Docs for AWE
- AirStripping0D unit model and related tests
- Docs for air stripping model
- Docs for basic water property package

### CODE REVIEW

File: src/watertap_contrib/reflo/property_models/air_water_equilibrium_properties.py
Comment (adam-a-a) [inline src/watertap_contrib/reflo/property_models/air_water_equilibrium_properties.py:1329]: Conventionally, I think these params would typically go on the param block. Any particular reason why you did it this way, other than convenience while developing? 

I suppose the only reason it matters is that one would expect to access parameters from the param block, and that'd be pretty consistent with most things in IDAES/WaterTAP, if my recollection is correct.

Comment (adam-a-a) [reply src/watertap_contrib/reflo/property_models/air_water_equilibrium_properties.py:1329]: I guess the other advantage of doing it this way is that parameters associated with on-demand props wouldn't be built unnecessarily, and building these on the param block would be less straight-forward if you want to also build on-demand.

Comment (kurbansitterley) [reply src/watertap_contrib/reflo/property_models/air_water_equilibrium_properties.py:1329]: Both of those reasons are pretty much it. 

And I suppose I'd push back a bit on this convention.  I don't see the advantage/logic behind putting them on the param block and them _always_ being built, but maybe I am missing something. It makes more sense to me to build property-specific params when the on-demand property is built. I would liken this to how we started putting unit-specific costing parameters only in unit-specific costing packages so they weren't always built and cluttering up calls to ``.display()`` or sifting through components via ``.component_objects()`` calls (maybe I am the only one that does that regularly though...)

But I definitely won't die on this hill. I am not opposed to putting them on the param block (or making them Vars, which appears to also be convention).

**Comment by kurbansitterley** [comment]
@dguittet any idea what is going on with these failing ``test_trough_surrogate`` tests?

**Comment by kurbansitterley** [comment]
> @dguittet any idea what is going on with these failing `test_trough_surrogate` tests?

Nevermind, not clear if it has anything to do with the surrogates. Same test fail as issue 89

File: src/watertap_contrib/reflo/property_models/air_water_equilibrium_properties.py
Comment (adam-a-a) [reply src/watertap_contrib/reflo/property_models/air_water_equilibrium_properties.py:1358]: I won't die on this hill either. There are mountains to climb! =)

File: src/watertap_contrib/reflo/costing/units/air_stripping.py
Comment (adam-a-a) [inline src/watertap_contrib/reflo/costing/units/air_stripping.py:390]: I would prefer this to be on the unit model itself. There could be the case where the user might only care for energy-related modeling without inclusion of costs.

File: src/watertap_contrib/reflo/costing/units/air_stripping.py
Comment (adam-a-a) [inline src/watertap_contrib/reflo/costing/units/air_stripping.py:376]: I had a similar comment for blower_power; I think any Var that could hypothetically be used without cost (e.g., energy/power consumption, etc.) should live on the unit model rather than in it's costing module.

File: src/watertap_contrib/reflo/costing/units/air_stripping.py
Comment (adam-a-a) [inline src/watertap_contrib/reflo/costing/units/air_stripping.py:560]: move to unit model

File: src/watertap_contrib/reflo/property_models/air_water_equilibrium_properties.py
Comment (adam-a-a) [inline src/watertap_contrib/reflo/property_models/air_water_equilibrium_properties.py:102]: Would benefit from some minimal documentation (doc strings)

File: src/watertap_contrib/reflo/costing/units/air_stripping.py
Comment (kurbansitterley) [reply src/watertap_contrib/reflo/costing/units/air_stripping.py]: Done

File: src/watertap_contrib/reflo/property_models/air_water_equilibrium_properties.py
Comment (kurbansitterley) [reply src/watertap_contrib/reflo/property_models/air_water_equilibrium_properties.py]: Done. Full documentation will be provided in readthedocs.

**Comment by adam-a-a** [comment]
I guess the question is--do we want this to go before or after #91? We can (1) merge #91 then see what needs to be updated here and fix, (2) merge this and see what needs to be fixed in #91, or (3) try merging your branch for #91 into this branch and proactively start fixing anything that breaks due to this prop model; that way, we could proceed with merging #91 and immediately merge this afterwards (instead of waiting for #91 to get merged in first as suggested in (1)).

**Review by adam-a-a — APPROVED** [review]

**Comment by kurbansitterley** [comment]
> I guess the question is--do we want this to go before or after #91? We can (1) merge #91 then see what needs to be updated here and fix, (2) merge this and see what needs to be fixed in #91, or (3) try merging your branch for #91 into this branch and proactively start fixing anything that breaks due to this prop model; that way, we could proceed with merging #91 and immediately merge this afterwards (instead of waiting for #91 to get merged in first as suggested in (1)).

I like (3), I will give it a try.

File: docs/technical_reference/unit_models/air_stripping.rst
Comment (adam-a-a) [inline docs/technical_reference/unit_models/air_stripping.rst:10]:
```suggestion
   * assumes isothermal and isobaric conditions
```

File: docs/technical_reference/unit_models/air_stripping.rst
Comment (adam-a-a) [inline docs/technical_reference/unit_models/air_stripping.rst:74]: Did you intend to add sets here?

File: src/watertap_contrib/reflo/unit_models/air_stripping_0D.py
Comment (adam-a-a) [inline src/watertap_contrib/reflo/unit_models/air_stripping_0D.py:270]: Minor, but any reason you moved away from calling a "safety factor"?

File: docs/technical_reference/unit_models/air_stripping.rst
Comment (kurbansitterley) [reply docs/technical_reference/unit_models/air_stripping.rst:74]: oops just added

File: src/watertap_contrib/reflo/unit_models/zero_order/tests/test_chemical_softening.py
Comment (zacharybinger) [inline src/watertap_contrib/reflo/unit_models/zero_order/tests/test_chemical_softening.py:66]: Can probably remove this now since not tested or make into an assert test

File: src/watertap_contrib/reflo/unit_models/air_stripping_0D.py
Comment (zacharybinger) [inline src/watertap_contrib/reflo/unit_models/air_stripping_0D.py:286]: Since you state the diameters for tower_port and tower_pipe in the doc, should bounds be added that reflect that? bound=(2,24)

**Review by zacharybinger — APPROVED** [review]
Small comments, but overall approved

File: src/watertap_contrib/reflo/unit_models/air_stripping_0D.py
Comment (kurbansitterley) [reply src/watertap_contrib/reflo/unit_models/air_stripping_0D.py:286]: I would but you can't add bounds to a ``Param``

File: src/watertap_contrib/reflo/unit_models/zero_order/tests/test_chemical_softening.py
Comment (kurbansitterley) [reply src/watertap_contrib/reflo/unit_models/zero_order/tests/test_chemical_softening.py:66]: oops good catch

**Pull request merged**
